src/HttpCache/EventListener/AddHeadersListener.php

```diff
@@ -70,7 +70,9 @@ public function onKernelResponse(FilterResponseEvent $event): void
             $response->setMaxAge($maxAge);
         }
 
-        if (null !== $this->vary) {
+        if (isset($resourceCacheHeaders['vary'])) {
+            $response->setVary($resourceCacheHeaders['vary']);
```

Comment:
This might result in duplicate `Vary` headers. Which is why 2 lines below we use `array_diff`. We should refactor this code a little to something like:

```php
$vary = $resourceCacheHeaders['vary'] ?? $this->vary;
if (null !== $vary) {
    $response->setVary(array_diff($vary, $response->getVary()), false);
}
```